chain/client/src/client_actor.rs

```diff
@@ -1280,6 +1280,12 @@ impl ClientActor {
                 .runtime_adapter
                 .get_epoch_height_from_prev_block(block.hash())
                 .unwrap_or(0);
+            let epoch_start_height = self
```

Comment:
This function does a tiny bit of work, and a whole lot of metrics work. Can that be moved to a separate function?
Also move the `last_final_stuff` into that function.